sequencer/src/persistence/fs.rs

```diff
@@ -222,41 +222,36 @@ impl Inner {
     fn collect_garbage(
         &mut self,
         view: ViewNumber,
-        intervals: &[RangeInclusive<u64>],
+        intervals: &[RangeInclusive<ViewNumber>],
```

Comment:
I can't work out why this function takes a mutable reference to self. Is it to avoid being called in two places at once?

Reply:
Yeah, I think that's the idea. It's not mutating the in-memory state that Rust cares about but it's still conceptually a mutable/exclusive function since it's updating shared files and directoreis